crates/services/p2p/src/peer_report.rs

```diff
@@ -0,0 +1,452 @@
+                        peer_id,
+                        handler,
+                        event: EitherOutput::Second(event),
+                    })
+                }
+                Poll::Ready(NetworkBehaviourAction::ReportObservedAddr {
+                    address,
+                    score,
+                }) => {
+                    return Poll::Ready(NetworkBehaviourAction::ReportObservedAddr {
+                        address,
+                        score,
+                    })
+                }
+                Poll::Ready(NetworkBehaviourAction::CloseConnection {
+                    peer_id,
+                    connection,
+                }) => {
+                    return Poll::Ready(NetworkBehaviourAction::CloseConnection {
+                        peer_id,
+                        connection,
+                    })
+                }
+                Poll::Ready(NetworkBehaviourAction::Dial { handler, opts }) => {
+                    let handler = IntoConnectionHandler::select(
+                        self.heartbeat.new_handler(),
+                        handler,
+                    );
+                    return Poll::Ready(NetworkBehaviourAction::Dial { handler, opts })
+                }
+                Poll::Ready(NetworkBehaviourAction::GenerateEvent(event)) => {
+                    match event {
+                        IdentifyEvent::Received {
+                            peer_id,
+                            info:
+                                IdentifyInfo {
+                                    protocol_version,
+                                    agent_version,
+                                    mut listen_addrs,
+                                    ..
+                                },
+                        } => {
+                            if listen_addrs.len() > MAX_IDENTIFY_ADDRESSES {
+                                debug!(
+                                    target: "fuel-p2p",
+                                    "Node {:?} has reported more than {} addresses; it is identified by {:?} and {:?}",
+                                    peer_id, MAX_IDENTIFY_ADDRESSES, protocol_version, agent_version
+                                );
+                                listen_addrs.truncate(MAX_IDENTIFY_ADDRESSES);
+                            }
+
+                            let event = PeerReportEvent::PeerIdentified {
+                                peer_id,
+                                agent_version,
+                                addresses: listen_addrs,
+                            };
+
+                            return Poll::Ready(NetworkBehaviourAction::GenerateEvent(
+                                event,
+                            ))
+                        }
+                        IdentifyEvent::Error { peer_id, error } => {
+                            debug!(target: "fuel-p2p", "Identification with peer {:?} failed => {}", peer_id, error)
```

Comment:
It's not a good idea to set the target to the wrong thing. The target should be this module not `fuel-core`